packages/ui/components/calendar/components/Calendar.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useMemo, useRef } from "react";
+
+import { useCalendarStore } from "../state/store";
+import "../styles/styles.css";
+import { CalendarComponentProps } from "../types/state";
+import { getDaysBetweenDates, getHoursToDisplay } from "../utils";
+import { DateValues } from "./DateValues";
+import { BlockedList } from "./blocking/BlockedList";
+import { EmptyCell } from "./event/Empty";
+import { EventList } from "./event/EventList";
+import { SchedulerColumns } from "./grid";
+import { SchedulerHeading } from "./heading/SchedulerHeading";
+import { HorizontalLines } from "./horizontalLines";
+import { VeritcalLines } from "./verticalLines";
+
+export function Calendar(props: CalendarComponentProps) {
+  const container = useRef<HTMLDivElement | null>(null);
+  const containerNav = useRef<HTMLDivElement | null>(null);
+  const containerOffset = useRef<HTMLDivElement | null>(null);
+  const initalState = useCalendarStore((state) => state.initState);
+
+  const { startDate } = useCalendarStore((state) => ({ startDate: state.startDate }));
```

Comment:
Wouldn't `const startDate = useCalendarStore(state => state.startDate)` be possible as well? Makes me think less because we won't have the creating of an object and the destructuring of it later.

Reply:
Good shout - we cal also move to the shallow approach and keep all of this in one object